### DISCUSSION TITLE
issue: Default value of array with empty string produces zero fields with useFieldArray

### DISCUSSION THREAD

**Discussion by IbrahimFadel** [react-hook-form/react-hook-form#10643] [Q&A] [answered]
### Version Number

7.43.9

### Codesandbox/Expo snack

https://codesandbox.io/p/sandbox/awesome-sky-vycxtf

### Steps to reproduce

1. initialize form default values with one field as an array with an empty string
2. use `useFieldArray` with the name equal to the field initialized in `defaultValues`
3. map `fields` returned by `useFieldArray` to text inputs
4. See no inputs, and no output from `console.log` in the map

### Expected behaviour

I want a growable list of text inputs (button to add/remove inputs) but to start with one input. I expected that by initializing the defaultValue to `[""]` it would do this

### What browsers are you seeing the problem on?

Firefox

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by leapful** [comment, marked as answer]
@IbrahimFadel 

react-hook-form does not support flat array field name like the one in your codesandbox sample `names.${index}`, it should be something like `names.${index}.value`

<img width="935" alt="Screenshot 2023-07-06 at 09 07 42" src="https://github.com/react-hook-form/react-hook-form/assets/18494222/99fe6cbd-20e9-487c-a7f0-fa4abdf9b90e">


Working demo: https://codesandbox.io/p/sandbox/intelligent-greider-cl3t29?file=%2Fapp%2Fpage.tsx%3A18%2C13

**Comment by IbrahimFadel** [comment]
Ah, thank you I did not see that line in the docs.
